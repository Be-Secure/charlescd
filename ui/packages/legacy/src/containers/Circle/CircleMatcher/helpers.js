import reduce from 'lodash/reduce'

export function buildParameters(params = {}) {
  return reduce(params, (acc, { key = '', value = '' }) => {
    acc[key] = value

    return acc
  }, {})
}

export function parseParameters(parameters) {
  try {
    const parsedParams = JSON.parse(parameters)

    return parsedParams
  } catch {
    return parameters
  }
}
