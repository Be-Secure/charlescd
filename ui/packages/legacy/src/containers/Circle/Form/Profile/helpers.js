import isUndefined from 'lodash/isUndefined'
import isArray from 'lodash/isArray'
import isEmpty from 'lodash/isEmpty'
import cloneDeep from 'lodash/cloneDeep'
import map from 'lodash/map'
import find from 'lodash/find'
import reduce from 'lodash/reduce'
import size from 'lodash/size'
import { responseTypes, initialRules, customerMapping } from './mocks'
import { ENGINE } from './constants'

export const FIRST_ELEMENT = 0

export const ONE = 1

export const isFirst = ruleId => ruleId === FIRST_ELEMENT

export const isClause = group => size(group) > ONE

export const isGroups = form => size(form.key) > ONE

export const hasAnyClause = form => find(form.key, isClause)

export const getFieldByKey = (key) => {
  const { type, values } = customerMapping[key] || {}
  const crmConditions = responseTypes[type] ?.conditions || []
  const crmValues = values

  return { crmConditions, crmValues }
}

function isValueSelect(key, engine) {
  const { crmValues } = getFieldByKey(key, customerMapping)

  return (ENGINE.REALWAVE === engine) && (size(crmValues) > ONE)
}

function getLogicalOperator(rules, groupId) {
  if (isUndefined(groupId)) {
    return rules.operator ? 'AND' : 'OR'
  }

  return rules.logicalOperator ?.[groupId] ? 'AND' : 'OR'
}

function convertClauseToForm(result, engine, group, groupId) {
  if (isUndefined(result.key[groupId])) {
    result.key[groupId] = []
    result.condition[groupId] = []
    result.ivalue[groupId] = []
    result.svalue[groupId] = []
  }

  map(group.clauses, ({ content }) => {
    result.key[groupId].push(content.key)
    result.condition[groupId].push(content.condition)

    if (isValueSelect(content.key, engine)) {
      result.ivalue[groupId].push(null)

      return result.svalue[groupId].push(content.value[FIRST_ELEMENT])
    }

    result.svalue[groupId].push(null)

    return result.ivalue[groupId].push(content.value[FIRST_ELEMENT])
  })

  result.logicalOperator.push(group.logicalOperator === 'AND')

  return result
}

export const convertToForm = (rules, engine) => {
  const profile = isEmpty(rules) ? initialRules : rules

  return (
    reduce(profile.clauses, (result, group, groupId) => {
      result.operator = profile.logicalOperator === 'AND'

      if (group.type === 'CLAUSE') {
        return convertClauseToForm(result, engine, group, groupId)
      }

      result.key[groupId] = [group.content.key]
      result.condition[groupId] = [group.content.condition]

      if (isValueSelect(group.content.key, engine)) {
        result.svalue[groupId] = [group.content.value[FIRST_ELEMENT]]
        result.ivalue[groupId] = [null]

        return result
      }

      result.svalue[groupId] = [null]
      result.ivalue[groupId] = [group.content.value[FIRST_ELEMENT]]

      return result
    }, { key: [], condition: [], ivalue: [], svalue: [], logicalOperator: [], operator: false })
  )
}

export const convertToProfile = (rules, engine) => {
  const clauses = map(rules.key, (group, groupId) => {
    const clauseRule = map(group, (rule, ruleId) => ({
      type: 'RULE',
      content: {
        key: rules.key[groupId][ruleId],
        value: isValueSelect(rules.key[groupId][ruleId], engine)
          ? [rules.svalue[groupId][ruleId]]
          : [rules.ivalue[groupId][ruleId]],
        condition: rules.condition[groupId][ruleId],
      },
    }))

    if (size(group) > ONE) {
      return {
        type: 'CLAUSE',
        clauses: clauseRule,
        logicalOperator: getLogicalOperator(rules, groupId),
      }
    }

    return clauseRule[FIRST_ELEMENT]
  })

  return {
    rules: {
      type: 'CLAUSE',
      clauses,
      logicalOperator: getLogicalOperator(rules),
    },
  }
}

export const removeRule = (form, groupId, ruleId) => {
  const rules = cloneDeep(form)
  const deleteCount = 1

  return reduce(rules, (result, value, key) => {
    if (key !== 'logicalOperator' && key !== 'operator') {
      value[groupId].splice(ruleId, deleteCount)
      if (isEmpty(value[groupId])) {
        value.splice(groupId, deleteCount)
      }
    }

    return result
  }, rules)
}

export const addNewGroup = form => ({
  ...form,
  key: [...form.key, [null]],
  condition: [...form.condition, [null]],
  ivalue: [...form.ivalue, [null]],
  svalue: [...form.svalue, [null]],
})

export const addNewRule = (form, groupId) => {
  const rules = cloneDeep(form)

  reduce(rules, (result, value) => {
    if (isArray(value[groupId])) {
      return value[groupId].push(null)
    }

    return result
  }, rules)

  if (isUndefined(rules.logicalOperator[groupId])) {
    return {
      ...rules,
      logicalOperator: [...rules.logicalOperator, false],
    }
  }

  return rules
}

export const changeValue = (form, field, value, groupId, ruleId) => {
  const newValue = Object.assign([], form[field][groupId], { [ruleId]: value })

  return {
    ...form,
    [field]: Object.assign(
      [],
      form[field],
      {
        [groupId]: newValue,
      },
    ),
  }
}
