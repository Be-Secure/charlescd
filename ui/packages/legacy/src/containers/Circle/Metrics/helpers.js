import map from 'lodash/map'
import first from 'lodash/first'
import { PERIODS } from './constants'

export const toList = data => map(data, ({ value, timestamp }) => [timestamp, value])

export const initialPeriodSelected = first(PERIODS)
