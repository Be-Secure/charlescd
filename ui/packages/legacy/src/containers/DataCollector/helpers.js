import reduce from 'lodash/reduce'

export const createReactSelect = (jsonFiles) => {
  return reduce(jsonFiles, (acc, jsonFile) => {
    const newValue = {
      value: jsonFile.metadata.name,
      label: jsonFile.metadata.name,
    }
    acc.push(newValue)

    return acc
  }, [])
}

export const formatSwitchText = field => field.split('.').join(' ')
