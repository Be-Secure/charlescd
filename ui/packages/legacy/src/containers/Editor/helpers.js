export const editorConfig = {
  toolbar: {
    buttons: [
      {
        name: 'h2',
        action: 'append-h3',
        aria: 'header type 2',
        tagNames: ['h3'],
        contentDefault: '<b>T</b>',
        classList: ['custom-class-h2'],
        attrs: {
          'data-custom-attr': 'attr-value-h2',
        },
      },
      'bold', 'italic', 'underline'],
  },
  paste: {
    forcePlainText: true,
    cleanPastedHTML: false,
    cleanReplacements: ['style'],
    cleanAttrs: ['class', 'style', 'dir'],
    cleanTags: ['meta', 'label'],
    unwrapTags: [],
  },
}

export function getElementByQuery(editor, query) {
  const [element] = editor.elements

  return element.querySelectorAll(query)
}

export function editorModeView(editor) {
  const [element] = editor.elements
  element.setAttribute('contentEditable', 'false')
}
