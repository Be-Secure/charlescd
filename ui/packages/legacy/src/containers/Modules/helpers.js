import map from 'lodash/map'

export const parseComponents = (components) => {
  return map(components, ({ name }) => ({ name }))
}
