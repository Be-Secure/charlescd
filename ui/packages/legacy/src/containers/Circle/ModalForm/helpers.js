import reduce from 'lodash/reduce'
import findIndex from 'lodash/findIndex'
import isEmpty from 'lodash/isEmpty'
import filter from 'lodash/filter'
import flatMap from 'lodash/flatMap'
import { ZERO, ONE } from 'core/helpers/constants'
import { SEGMENTS_TYPE, MATCHER_TYPE } from './constants'

export const groupModulesById = (modules) => {
  return reduce(modules, (result, module) => {
    const index = findIndex(result, ({ id }) => id === module.id)
    if (index >= ZERO) {
      return Object.assign([], result, {
        [index]: {
          id: module.id,
          components: [
            ...result[index].components,
            ...module.components,
          ],
        },
      })
    }

    return [...result, module]
  }, [])
}

export const filterDataByName = (items, value) => {
  return isEmpty(value)
    ? items
    : filter(items, ({ name }) => name.toLowerCase().includes(value.toLowerCase()))
}

export const checkComponentsDuplicate = (modules, component) => {
  const duplicates = filter(
    flatMap(modules, mod => mod?.components), ({ id }) => id === component?.id,
  )

  return duplicates?.length > ONE
}

export function getAttentionMessage(matcherType, segmentType) {
  const isManualMessage = (
    matcherType === MATCHER_TYPE.SIMPLE_KV && segmentType === SEGMENTS_TYPE.MANUALLY
  )

  if (isManualMessage) {
    return 'circle.attention.message.manual'
  }

  return 'circle.attention.message.import'
}
