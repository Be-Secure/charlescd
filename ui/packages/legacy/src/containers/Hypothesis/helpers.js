import { getRangeWords, removeHtmlTags } from 'core/helpers/regex'

export const shortDescription = description => (
  getRangeWords(removeHtmlTags(description), '...')
)
