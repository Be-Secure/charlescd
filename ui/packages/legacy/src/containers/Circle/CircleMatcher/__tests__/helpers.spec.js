import { buildParameters, parseParameters } from '../helpers'
import { params } from './fixtures'

describe('Circle Matcher helpers', () => {
  test('should build parameters', () => {
    const buildedParams = buildParameters(params)

    expect(buildedParams).toMatchSnapshot()
  })

  test('Should build parameters with empty values', () => {
    expect(buildParameters({})).toEqual({})
    expect(buildParameters(null)).toEqual({})
    expect(buildParameters(undefined)).toEqual({})
  })

  test('Should parse parameters', () => {
    const parameters = '{ "foo": "bar" }'
    const parsedParams = parseParameters(parameters)
    expect(parsedParams).toMatchObject({ foo: 'bar' })
  })

  test('Should try to parse an invalid parameters', () => {
    const invalidJSON = 'foo{bar'
    const parsedParams = parseParameters(invalidJSON)
    expect(parsedParams).toBe(invalidJSON)
  })
})
