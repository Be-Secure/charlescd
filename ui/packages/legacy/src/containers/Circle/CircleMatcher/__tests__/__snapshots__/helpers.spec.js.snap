// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Matcher helpers should build parameters 1`] = `
Object {
  "": "value",
  "email": "",
  "name": "John Wick",
  "password": "12345",
  "user": "foobar",
}
`;
